website/static/js/filepage/index.js

```diff
@@ -157,6 +157,71 @@ var FileViewPage = {
         }, 1000);
 
         self.mfrIframeParent = $('#mfrIframeParent');
+
+        $.fn.editable.defaults.mode = 'inline';
+
+        if(self.canEdit) {
+            var $fileName = $('#fileName');
+            var conflict = 'warn';
+            var to = {
+                data: {
+                    nodeId: self.node.id,
+                    path: null,
+                    provider: self.file.provider,
+                }
+            };
+            var from = JSON.parse(JSON.stringify(to));
+            from.data.path = self.file.path;
+            $fileName.editable({
+                type: 'text',
+                send: 'always',
+                url: waterbutler.moveUrl(),
+                ajaxOptions:{
+                    beforeSend: $osf.setXHRAuthorization,
+                    headers: {
+                        'Content-Type': 'Application/json'
+                    },
+                },
+                params: function(params) {
+                    params = JSON.stringify({
+                        'rename': params.value,
+                        'conflict': conflict,
+                        'source': waterbutler.toJsonBlob(from),
+                        'destination': waterbutler.toJsonBlob(to),
+                    });
+                    return params;
+                },
+                validate: function(value) {
+                    if($.trim(value) === ''){
+                        $osf.growl('Error', 'Name cannot be empty.');
+                        return '   ';
+                    } else if(value.length > 100){
+                        $osf.growl('Error', 'Name should be less than 100 characters');
+                        return '   ';
+                    }
+                },
+                success: function(response, value) {
+                    window.location.reload();
+                },
+                error: function(response) {
+                    var msg = response.responseJSON.message;
+                    if (msg) {
+                        $osf.growl('Error', msg);
+                        return '      ';
```

Comment:
Why does this return a blank string here instead of `null`? (and with a different number of spaces than other branches)

When returning no value, be explicit if possible. This increases clarity and avoids unexpected side-effects.